kubetest2/internal/deployers/eksapi/nodegroup.go

```diff
@@ -532,3 +533,72 @@ func (m *NodegroupManager) getSubnetWithCapacity(infra *Infrastructure, opts *de
 	klog.Infof("Using subnet: %s", subnetId)
 	return subnetId, capacityReservationId, nil
 }
+
+func (m *NodegroupManager) getValidDefaultInstanceTypesByEKSAMIType(amiType ekstypes.AMITypes) ([]string, error) {
+	defaults, ok := defaultInstanceTypesByEKSAMITypes[amiType]
+	if !ok {
+		return []string{}, fmt.Errorf("no default instance types known for AmiType: %v", amiType)
+	}
+
+	return m.getValidInstanceTypesFromList(defaults), nil
+}
+
+func (m *NodegroupManager) getValidDefaultInstanceTypesByEC2Arch(arch ec2types.ArchitectureValues) ([]string, error) {
+	defaults, ok := defaultInstanceTypesByEC2ArchitectureValues[arch]
+	if !ok {
+		return []string{}, fmt.Errorf("no default instance types known for AMI architecture: %v", arch)
+	}
+
+	return m.getValidInstanceTypesFromList(defaults), nil
+}
+
+func (m *NodegroupManager) getValidInstanceTypesFromList(desiredInstanceTypes []ec2types.InstanceType) []string {
+	_, err := m.clients.EC2().DescribeInstanceTypes(context.TODO(), &ec2.DescribeInstanceTypesInput{
```

Comment:
You should be able to check the error code here like:
```
if err != nil {
	var apierr smithy.APIError
	if errors.As(err, &apierr) {
		// do something with apierr.ErrorCode()
    }
}
```
instead of doing the parsing yourself

Reply:
Just pushed a change to parse the error code in this format and pass just the error message to the invalid instance types parser rather than the whole error.